src/bootstrap/src/lib.rs

```diff
@@ -490,18 +490,30 @@ impl Build {
             return;
         }
 
-        let submodule_git = || helpers::git(Some(&absolute_path));
+        // Submodule updating actually happens during in the dry run mode. We need to make sure that
+        // all the git commands below are actually executed, because some follow-up code
+        // in bootstrap might depend on the submodules being checked out. Furthermore, not all
+        // the command executions below work with an empty output (produced during dry run).
+        // Therefore, all commands below are marked with `run_always()`, so that they also run in
+        // dry run mode.
+        let submodule_git = || {
+            let mut cmd = helpers::git(Some(&absolute_path)).capture_stdout();
+            cmd.run_always();
+            cmd
+        };
 
         // Determine commit checked out in submodule.
-        let checked_out_hash = output(submodule_git().args(["rev-parse", "HEAD"]).as_command_mut());
+        let checked_out_hash =
+            submodule_git().run_always().args(["rev-parse", "HEAD"]).run(self).stdout();
```

Comment:
```suggestion
            submodule_git().args(["rev-parse", "HEAD"]).run(self).stdout();
```
the `run_always()` calls on `submodule_git()` are redundant?

Reply:
Yes, this was a leftover. Good catch, thanks! Fixed.